sockeye/inference.py

```diff
@@ -1535,6 +1544,7 @@ def _beam_search(self,
                 if mx.nd.sum(finished).asscalar() > 0:
                     break
             else:
+                finished = mx.nd.sign(finished + (mx.nd.cast(lengths.reshape(-1), 'int32') > max_output_lengths))
```

Comment:
I think this check should actually be `lengths >= max_output_lengths`. For a single translation at the end of the for loop lengths will be exactly equal to max_output_length (because it starts at 1).